test/puppetlabs/puppetdb/testutils/jetty.clj

```diff
@@ -60,16 +60,16 @@
     (spit logback-file (log-config log-file "ERROR"))
     [log-file (assoc-in config [:global :logging-config] logback-file)]))
 
-(defn current-port
-  "Given a trapperkeeper server, return the port of the running jetty instance.
-  Note there can be more than one port (i.e. SSL + non-SSL connector). This only
-  returns the first one."
-  [server]
-  (-> @(tka/app-context server)
-      (get-in [:WebserverService :jetty9-servers :default :server])
-      .getConnectors
-      first
-      .getLocalPort))
+(defn open-port-num
+  "Returns a currently open port number"
+  []
+  (with-open [s (java.net.ServerSocket. 0)]
```

Comment:
This is the same as Rob's concern below, I see now. I suppose such things are unlikely, and that we should probably just go with it. We can re-evaluate if we start seeing port evaluation problems in test runs.